fern/pages/src/providers/py-custom.mdx

```diff
@@ -1,6 +1,540 @@
 ---
-title: Python Custom Provider
+title: Python custom provider
 image: "https://og.composio.dev/api/og?title=Python%20Custom%20Provider"   # image for socials
 keywords: ""
 hide-nav-links: false
 ---
+
+This guide shows how to create custom providers for the Composio Python SDK. Custom providers enable integration with different AI frameworks and platforms.
+
+## Provider architecture
+
+The Composio SDK uses a provider architecture to adapt tools for different AI frameworks. The provider handles:
+
+1. **Tool format transformation**: Converting Composio tools into formats compatible with specific AI platforms
+2. **Tool execution**: Managing the flow of tool execution and results
+3. **Platform-specific integration**: Providing helper methods for seamless integration
+
+## Types of providers
+
+There are two types of providers:
+
+1. **Non-agentic providers**: Transform tools for platforms that don't have their own agency (e.g., OpenAI, Anthropic)
+2. **Agentic providers**: Transform tools for platforms that have their own agency (e.g., LangChain, CrewAI)
+
+## Provider class hierarchy
+
+```
+BaseProvider (Abstract)
+├── NonAgenticProvider (Abstract)
+│   └── OpenAIProvider (Concrete)
+│   └── AnthropicProvider (Concrete)
+│   └── [Your Custom Non-Agentic Provider] (Concrete)
+└── AgenticProvider (Abstract)
+    └── LangchainProvider (Concrete)
+    └── [Your Custom Agentic Provider] (Concrete)
+```
+
+## Quick start
+
+The fastest way to create a new provider is using the provider scaffolding script:
+
+```bash
+# Create a non-agentic provider
+make create-provider name=myprovider
+
+# Create an agentic provider
+make create-provider name=myagent agentic=true
+```
+
+This will create a new provider in `python/providers/<provider-name>/` with:
+- Complete package structure with `pyproject.toml` and `setup.py`
+- Provider implementation template
+- Demo script
+- README with usage examples
+- Type annotations and proper inheritance
```

Comment:
Ideally we should allow users to maintain the providers on their own, this indicates that the provider implementation needs to be submitted to composio repo